pkg/sql/schemachanger/rel/database.go

```diff
@@ -145,6 +159,78 @@ func NewDatabase(sc *Schema, indexes ...Index) (*Database, error) {
 	return t, nil
 }
 
+// maybeExpandInvertedIndexes will replace inverted indexes defined by the
+// user with two new indexes corresponding to the inverted attribute in
+// order to enable efficient lookup by either source or value. This is
+// done because generally in rel it is safe to assume that it is efficient
+// to go from an entity to its attribute and that it is also efficient to go
+// from an attribute to the set of entities so long as the attribute is
+// indexed. Because the slice members are not directly a part of the entity
+// in the rel model, we need to index from the entity to the slice member to
+// make that expansion cheap.
+func maybeExpandInvertedIndexes(sc *Schema, indexes []Index) ([]Index, error) {
+	if !containsInverted(indexes) {
+		return indexes, nil
+	}
+	var expanded []Index
+	for _, idx := range indexes {
+		if !idx.Inverted {
+			expanded = append(expanded, idx)
+			continue
+		}
+		if err := validatedInvertedIndexProperties(sc, idx); err != nil {
+			return nil, err
+		}
+		expanded = append(expanded, Index{
+			Attrs:    []Attr{idx.Attrs[0], sliceSource, sliceIndex},
```

Comment:
maybe a redundant commentary here that there'll be only one attribute if it's inverted index so it's safe to just do `idx.Attrs[0]`